Bottom: 73b080c9773fbc8d8b48f2264a550509cd8dbb21
Top:    65d69d3d6aceb0d94b70bc43e3652002edea0f52
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-03 23:46:22 +0000

Try using cvc_next instead of cvc_prev where possible


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 77e0282..44271ce 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -254,7 +254,7 @@ new_descriptor (VNODE * a, char poly, char side)
     {
       if (side == 'P')
 	{
-	  if (PREV_VERTEX (a)->cvc_prev == (CVCList *) - 1)
+	  if (PREV_VERTEX (a)->cvc_next == (CVCList *) - 1)
 	    PREV_VERTEX (a)->cvc_prev = PREV_VERTEX (a)->cvc_next = NULL;
 	  poly_ExclVertex (PREV_VERTEX (a));
           fprintf (stderr, "REMOVING VERTEX 111111111111111111111111111111111111111111111111111111\n");
@@ -263,7 +263,7 @@ new_descriptor (VNODE * a, char poly, char side)
 	}
       else
 	{
-	  if (NEXT_VERTEX (a)->cvc_prev == (CVCList *) - 1)
+	  if (NEXT_VERTEX (a)->cvc_next == (CVCList *) - 1)
 	    NEXT_VERTEX (a)->cvc_prev = NEXT_VERTEX (a)->cvc_next = NULL;
 	  poly_ExclVertex (NEXT_VERTEX (a));
           fprintf (stderr, "REMOVING VERTEX 222222222222222222222222222222222222222222222222222222\n");
@@ -601,7 +601,7 @@ add_descriptors (PLINE * pl, char poly, CVCList * list)
 
   do
     {
-      if (node->cvc_prev)
+      if (node->cvc_next)
 	{
 	  assert (node->cvc_prev == (CVCList *) - 1
 		  && node->cvc_next == (CVCList *) - 1);
@@ -1703,7 +1703,7 @@ jump (VNODE **curv, DIRECTION *cdir, J_Rule j_rule)
   VNODE *e; /* e is considered an edge */
   DIRECTION newone;
 
-  if (!(*curv)->cvc_prev)	/* not a cross-vertex */
+  if (!(*curv)->cvc_next)	/* not a cross-vertex */
     {
       if (VERTEX_DIRECTION_EDGE (*curv, *cdir)->Flags.mark)
 	return FALSE;
